web/client/src/components/Credentials.js

```diff
@@ -0,0 +1,122 @@
+import React, { useState } from "react";
+import Button from "./Button";
+import { MDBContainer, MDBRow, MDBCol, MDBBtn } from "mdbreact";
+import Radio from "./Radio";
+
+function Credentials() {
+  const [type, setType] = useState("");
+  const [format, setFormat] = useState("");
+  const formatChange = (e) => setFormat(e.target.value); // refactoring code
+  const typeChange = (e) => setType(e.target.value); // refactoring code
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    if (e.target.format.value === "Header" && e.target.type.value === "JWT") {
+      alert("JWT type and Header format are not allowed!");
+    } else {
+      console.log(e.target.type.value);
+      console.log(e.target.format.value);
+    }
+  };
+
+  return (
+    <div className="top">
+      <div className="shadow-box-example z-depth-2">
+        <MDBContainer>
+          <form onSubmit={handleSubmit}>
+            <MDBCol>
+              {" "}
+              <p className="h5 text-center mb-4">Choose token type </p>
+            </MDBCol>
+            <fieldset id="type">
+              <MDBRow>
+                <MDBCol>
+                  <Radio
+                    name="type"
+                    value="OAuth"
+                    id="defaultGroupExample1"
+                    onChange={typeChange}
+                  />
+                </MDBCol>
+                <MDBCol>
+                  {" "}
+                  <Radio
+                    name="type"
+                    value="JWT"
+                    id="defaultGroupExample2"
+                    onChange={typeChange}
+                  />
+                </MDBCol>
+              </MDBRow>
+            </fieldset>
+
+            <MDBCol>
+              {" "}
+              <p className="h5 text-center mb-4">Choose token format </p>
+            </MDBCol>
+            <fieldset id="format">
+              <MDBRow>
+                <MDBCol>
+                  {" "}
+                  <Radio
+                    name="format"
+                    value="Bare"
+                    id="defaultGroupExample3"
+                    onChange={formatChange}
```

Comment:
You may be able to just have `onChange={setFormat}` here and for other cases where you use `onChange` with `formatChange` or `typeChange`

Reply:
Yea I was able to, thanks!